File: skills/astra-toolkit/clients/csharp/examples/tables-list-types.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      System.Environment.GetEnvironmentVariable("ASTRA_DB_API_ENDPOINT"),
      System.Environment.GetEnvironmentVariable("ASTRA_DB_APPLICATION_TOKEN")
    );

    // List types
    var types = await database.ListTypesAsync();

    Console.WriteLine(JsonSerializer.Serialize(types));
  }
}

File: skills/astra-toolkit/clients/java/examples/tables-list-types.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.definition.types.TableUserDefinedTypeDescriptor;
import java.util.List;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient(System.getenv("ASTRA_DB_APPLICATION_TOKEN")).getDatabase(System.getenv("ASTRA_DB_API_ENDPOINT"));

    // List type metadata
    List<TableUserDefinedTypeDescriptor> result = database.listTypes();

    System.out.println(result);
  }
}
